src/main/java/net/minestom/server/utils/mojang/MojangUtils.java

```diff
@@ -90,6 +102,29 @@ public final class MojangUtils {
         }
     }
 
+    @Blocking
+    @ApiStatus.Internal
+    public static @Nullable JsonObject authenticateSession(String loginUsername, String serverId, @Nullable SocketAddress userSocket) {
+        final String username = URLEncoder.encode(loginUsername, StandardCharsets.UTF_8);
+
+        final String url;
+        if (ServerFlag.AUTH_PREVENT_PROXY_CONNECTIONS
+                && userSocket instanceof InetSocketAddress inetSocketAddress
+                && inetSocketAddress.getAddress() instanceof InetAddress address
+        ) {
+            url = String.format(PREVENT_PROXY_CONNECTIONS_AUTH_URL, username, serverId, address.getHostAddress());
+        } else {
+            url = String.format(BASE_AUTH_URL, username, serverId);
+        }
+
+        try {
+            return retrieve(url);
+        } catch (IOException e) {
+            MinecraftServer.getExceptionManager().handleException(e);
```

Comment:
Would prefer if you just threw a `RuntimeException`. The exception manager is already used in the listener

Reply:
Do you want the Login Listener to handle the exception, or should we let it be managed by the main packet listener try-catch?

If we handle it in the Login Listener, it might be best to let it throw an `IOException`. This way, we can use custom error handling for it, treating it as a Mojang error.

Reply:
In the listener seems fine. Also maybe explicitly define in `MojangUtils` signature the exception thrown